hydromt_wflow/workflows/forcing.py

```diff
@@ -73,3 +99,170 @@ def pet(
+        smoothing. Default is 0.
+    minimum_neighbours : int, optional
+        Minimum number of neighbors needed to perform Barnes or Cressman interpolation
+        for a point. Default is 3.
+    search_radius : float, optional
+        A search radius to use for the Barnes and Cressman interpolation schemes.
+        If search_radius is not specified, it will default to 5 times the average
+        spacing of observations.
+    gamma : float, optional
+        Adjustable smoothing parameter for the barnes interpolation. Default is 0.25.
+    kappa_star : float, optional
+        Response parameter for barnes interpolation, specified nondimensionally
+        in terms of the Nyquist. Default is 5.052.
+
+    Returns
+    -------
+    xr.DataArray
+        Interpolated forcing data on a regular grid with dimensions (time, y, x).
+
+    See Also
+    --------
+    hydromt_wflow.workflows.forcing.spatial_interpolation
+    """
+    if not HAS_METPY:
+        raise ModuleNotFoundError("MetPy package is required for spatial interpolation")
+
+    if np.isnan(forcing.values).any():
+        logger.warning(
+            """Forcing data contains NaN values. \
+These will be skipped during interpolation.""")
+
+    if interp_type not in interpolation_supported.keys():
+        raise ValueError(f"Interpolation type {interp_type} not recognized.")
+
+    elif interpolation_supported[interp_type]:
+        # Create a dictionary of the arguments to show in logging
+        interp_args_dict = {
+            "rbf_func": rbf_func,
+            "rbf_smooth": rbf_smooth,
+            "minimum_neighbors": minimum_neighbours,
+            "search_radius": search_radius,
+            "gamma": gamma,
+            "kappa_star": kappa_star,
+        }
+
+        # Filter the arguments based on the interpolation type
+        interp_args = ", ".join(
+            [
+                f"{key}={interp_args_dict[key]}"
+                for key in interpolation_supported[interp_type]
+            ]
+        )
+        logger.info(
+            f"Starting interpolation with interp_type '{interp_type}' ({interp_args})."
+        )
+
+    else:
+        logger.info(f"Starting interpolation with interp_type '{interp_type}'.")
+
+    # Determine the dimensions of the resulting interpolated grid
+    x = stations.geometry.x
+    y = stations.geometry.y
+    boundary_coords = get_boundary_coords(x, y)
+    grid_x, grid_y = generate_grid(hres, boundary_coords)
+    
+    # Create empty DataArray to store the interpolated values
+    coords = {"time": forcing.index, "x": grid_x[0, :], "y": grid_y[:, 0]}
+    da_forcing = xr.DataArray(coords=coords, dims=["time", "y", "x"])
+    da_forcing.raster.set_crs(stations.crs)
+    
+    empty_timesteps = []
+    last_logged_progress = 0
+    for i, (timestep, observations) in enumerate(forcing.iterrows()):
+        
+        # TODO is there a nicer way using the logger?
+        progress = ((i + 1)/len(forcing))*100
```

Comment:
I do not mind so much, this is actually quite elegant :) else we can install the additional tqdm package but this means more dependencies once again so not sure how worth it it would be.